spark/src/main/java/org/apache/iceberg/spark/data/SparkOrcReader.java

```diff
@@ -0,0 +1,823 @@
+import org.apache.orc.storage.ql.exec.vector.StructColumnVector;
+import org.apache.orc.storage.ql.exec.vector.TimestampColumnVector;
+import org.apache.orc.storage.ql.exec.vector.VectorizedRowBatch;
+import org.apache.orc.storage.serde2.io.DateWritable;
+import org.apache.orc.storage.serde2.io.HiveDecimalWritable;
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.catalyst.expressions.SpecializedGetters;
+import org.apache.spark.sql.catalyst.expressions.UnsafeRow;
+import org.apache.spark.sql.catalyst.expressions.codegen.UnsafeArrayWriter;
+import org.apache.spark.sql.catalyst.expressions.codegen.UnsafeRowWriter;
+import org.apache.spark.sql.catalyst.expressions.codegen.UnsafeWriter;
+import org.apache.spark.sql.catalyst.util.ArrayData;
+import org.apache.spark.sql.catalyst.util.MapData;
+import org.apache.spark.sql.types.Decimal;
+import org.apache.spark.unsafe.Platform;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.sql.Timestamp;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Converts the OrcInterator, which returns ORC's VectorizedRowBatch to a
+ * set of Spark's UnsafeRows.
+ *
+ * It minimizes allocations by reusing most of the objects in the implementation.
+ */
+public class SparkOrcReader implements Iterator<InternalRow>, Closeable {
+  private final static int INITIAL_SIZE = 128 * 1024;
+  private final OrcIterator reader;
+  private final UnsafeRowWriter writer;
+  private int nextRow = 0;
+  private VectorizedRowBatch current = null;
+  private Converter[] converter;
+
+  public SparkOrcReader(InputFile location,
+                        FileScanTask task,
+                        Schema readSchema) {
+    ColumnIdMap columnIds = new ColumnIdMap();
+    TypeDescription orcSchema = TypeConversion.toOrc(readSchema, columnIds);
+    reader = ORC.read(location)
+        .split(task.start(), task.length())
+        .schema(readSchema)
+        .build();
+    int numFields = readSchema.columns().size();
+    writer = new UnsafeRowWriter(numFields, INITIAL_SIZE);
+    converter = new Converter[numFields];
+    for(int c = 0; c < numFields; ++c) {
+      converter[c] = buildConverter(writer, orcSchema.getChildren().get(c));
+    }
+  }
+
+  @Override
+  public boolean hasNext() {
+    return (current != null && nextRow < current.size) || reader.hasNext();
+  }
+
+  @Override
+  public UnsafeRow next() {
+    if (current == null || nextRow >= current.size) {
+      current = reader.next();
+      nextRow = 0;
+    }
+
+    writer.reset();
+    writer.zeroOutNullBytes();
+    for(int c=0; c < current.cols.length; ++c) {
+      converter[c].convert(writer, c, current.cols[c], nextRow);
+    }
+    nextRow++;
+    return writer.getRow();
+  }
+
+  @Override
+  public void close() throws IOException {
+    reader.close();
+  }
+
+  private static void printRow(SpecializedGetters row, TypeDescription schema) {
```

Comment:
I understand wanting to see data for debugging, but this should be done using a logger and conversion methods to produce strings.

Reply:
Yeah, I agree with you. Since this code was already merged in the Netflix repo: https://github.com/Netflix/iceberg/blob/master/spark/src/main/java/com/netflix/iceberg/spark/data/SparkOrcReader.java#L122 and https://github.com/Netflix/iceberg/blob/master/spark/src/main/java/com/netflix/iceberg/spark/data/SparkOrcReader.java#L135 I assumed it was alright, but I should've check before again. Thanks for caching this, I'll remove any print statements.

Reply:
Sorry, I didn't realize it was part of the original! It's good to have some of this debugging code, but we should put it in the right places.